src/main/php/com/mongodb/io/Compression.class.php

```diff
@@ -0,0 +1,98 @@
+<?php namespace com\mongodb\io;
+
+use lang\Value;
+use util\Comparison;
+
+/** 
+ * Compression negotiation and selection.
+ * 
+ * @see   https://github.com/mongodb/specifications/blob/master/source/compression/OP_COMPRESSED.md
+ * @test  com.mongodb.unittest.CompressionTest
+ */
+class Compression implements Value {
+  use Comparison;
+
+  private static $negotiable= [];
+  private $compressors;
+
+  static function __static() {
+    extension_loaded('zlib') && self::$negotiable['zlib']= fn($options) => new Zlib($options['zlibCompressionLevel'] ?? -1);
+    extension_loaded('zstd') && self::$negotiable['zstd']= fn($options) => new Zstd();
```

Comment:
The spec doesn't list a `zstdCompressionLevel` flag, although we might decide to add it

Reply:
The golang driver has this flag, [see here](https://github.com/mongodb/mongo-go-driver/blob/056ba3c1e2885d818eb6fd0f15baba8115cdcc64/x/mongo/driver/connstring/connstring_spec_test.go#L268)